### PR TITLE
[HttpKernel] Remove bundle inheritance

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Blocked by symfony/symfony#24160

### CODE REVIEW

**Comment by xabbuh** [comment]
The TwigBundle also deals with the bundle hierarchy when resolving paths for bundle templates: https://github.com/fabpot/symfony/blob/218626157e915f4e26cc8bb619637248c0008c72/src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php#L164-L214 I bet we can simplify a lot of code there too.

**Comment by fabpot** [comment]
@xabbuh now removed

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php [hunk 1]
```diff
@@ -169,55 +160,21 @@ public function load(array $configs, ContainerBuilder $container)
     private function getBundleHierarchy(ContainerBuilder $container, array $config)
```
Comment (yceruto) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: This inner code can be moved out safely, there is not much logic to encapsulate now.

Comment (fabpot) [reply src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: Not sure, I prefer to keep it.

Comment (javiereguiluz) [reply src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: Could we update the method name? Maybe `getBundleTemplatePaths()` or something like that?

Comment (fabpot) [reply src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: done

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php [hunk 2]
```diff
@@ -166,58 +157,24 @@ public function load(array $configs, ContainerBuilder $container)
         $container->registerForAutoconfiguration(RuntimeExtensionInterface::class)->addTag('twig.runtime');
     }
 
-    private function getBundleHierarchy(ContainerBuilder $container, array $config)
+    private function getBundleTemplatePaths(ContainerBuilder $container, array $config)
     {
         $bundleHierarchy = array();
```
Comment (inso) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: Maybe variable should be renamed in conformity with method name change?

**Comment by fabpot** [comment]
Tests are green now!

File: src/Symfony/Component/HttpKernel/Tests/KernelTest.php [hunk 3]
```diff
@@ -535,129 +414,6 @@ public function testLocateResourceOnDirectories()
...
-        $child2 = $this->getBundle(null, 'ParentCBundle', 'ChildC2Bundle');
-
-        $kernel = $this->getKernel(array(), array($parent, $child1, $child2));
-        $kernel->boot();
-    }
-
-    /**
-     * @group legacy
-     * @expectedException \LogicException
-     * @expectedExceptionMessage Trying to register two bundles with the same name "DuplicateName"
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: that test is still relevant and should be kept as it is not related to bundle inheritance

Comment (fabpot) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: Indeed, I will fix it in 3.4 as this should not be deprecated there.

Comment (Tobion) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: I think you forgot to remove the legacy annotation in 3.4

Comment (xabbuh) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: The `@group` annotation will be removed in #24364.

Comment (Tobion) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: #24364 is for master. I don't see how this is related to this 3.4 issue.

Comment (xabbuh) [reply src/Symfony/Component/HttpKernel/Tests/KernelTest.php]: The test fails with a deprecation on 3.4 otherwise.

**Pull request merged**